core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -168,6 +170,26 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
           driverConf.set(key, value)
         }
       }
+      // Delegation Token Updater is not supported in Hadoop 1, so use reflection.
+      // Can't use Option[ExecutorDelegationTokenUpdater] because it is built only in YARN
+      // profile, so use Option[Any] since even the stop method call will be via reflection.
+      var tokenUpdaterOption: Option[Any] = None
+      var tokenUpdaterClass: Option[Class[_]] = None
```

Comment:
I didn't want to add more methods to `SparkHadoopUtil` and `YarnSparkHadoopUtil`, but since that gets rid of the reflection and also guarantees that `addCredentials` gets called only in YARN mode, I guess it is worth it.